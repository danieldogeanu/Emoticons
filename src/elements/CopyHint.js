import React from 'react';

function CopyHint(props) {
	return (
		<span className="CopyHint">
			{props.text}
		</span>
	);
}

export default CopyHint;
