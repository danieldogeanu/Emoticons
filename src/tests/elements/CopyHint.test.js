import React from 'react';
import {render} from '@testing-library/react';
import CopyHint from '../../elements/CopyHint';

describe('CopyHint Element', () => {

	it('renders copy hint element properly', () => {
		const testHint = 'Test Hint';
		const {getByText} = render(<CopyHint text={testHint} />);
		const renderedHint = getByText(testHint);

		expect(renderedHint).toBeInTheDocument();
		expect(renderedHint).toHaveClass('CopyHint');
		expect(renderedHint).toHaveTextContent(testHint);
		expect(renderedHint).toMatchSnapshot();
	});

});
