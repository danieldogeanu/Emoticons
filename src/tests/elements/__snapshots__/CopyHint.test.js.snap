// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyHint Element renders copy hint element properly 1`] = `
<span
  class="CopyHint"
>
  Test Hint
</span>
`;
